docs/src/pages/components/pagination/examples/PaginationHasMorePagesExample.tsx

```diff
@@ -0,0 +1,64 @@
+import { useState } from 'react';
+import { Pagination } from '@aws-amplify/ui-react';
+
+export const PaginationHasMorePagesExample = () => {
+  const [pageTokens, setPageTokens] = useState([
+    {
+      res: 'dog',
+      nextToken: '123',
+    },
+  ]);
+  const [currPageIndex, setCurrPageIndex] = useState(1);
+  const [hasMorePages, setHasMorePages] = useState(true);
+
+  const handleNextPage = async () => {
+    if (hasMorePages && currPageIndex === pageTokens.length) {
+      const resultsFromAPI = await mockedAPI();
+
+      if (!resultsFromAPI.nextToken) {
+        setHasMorePages(false);
+      }
+
+      setPageTokens([...pageTokens, resultsFromAPI]);
+    }
+
+    setCurrPageIndex(currPageIndex + 1);
+  };
+
+  const handlePreviousPage = () => {
+    if (currPageIndex > 1) {
+      setCurrPageIndex(currPageIndex - 1);
+    }
+  };
+
+  return (
+    <Pagination
+      currentPage={currPageIndex}
+      totalPages={pageTokens.length}
+      hasMorePages={hasMorePages}
+      onNext={handleNextPage}
+      onPrevious={handlePreviousPage}
+    />
+  );
+};
+
+const mockedAPI = (function () {
+  const response = [
+    {
+      res: 'cat',
+      nextToken: '456',
+    },
+    {
+      res: 'parrot',
+      nextToken: null,
+    },
+  ];
+
+  let timesCalled = 0;
+
+  return function () {
+    if (timesCalled < 3) {
```

Comment:
```suggestion
    if (timesCalled < 2) {
```

Looks like you only got 2 items in the array

Reply:
Good catch! I originally had 3 items in the array, but I moved the first item into `pageTokens`:
```jsx
  const [pageTokens, setPageTokens] = useState([
    {
      res: 'dog',
      nextToken: '123',
    },
  ]);
  ```

I tried messing around with `useEffect` to load the first item on the initial render but ran into some issues.